jobs/periodic/deactivate-missing-coc/index.ts

```diff
@@ -0,0 +1,43 @@
+import {getLogger} from "log4js";
+import { prisma } from "../../../common/prisma";
+import validator from "validator";
+import toDate = validator.toDate;
+import {deactivateStudent} from "../../../common/student/activation";
+
+const logger = getLogger();
+
+export default async function execute() {
+    logger.info("Missing CoC deactivation job: looking for students wo have not supplied CoC within 8 Weeks after registration...");
+    await deactivateMissingCoc();
+}
+
+export async function deactivateMissingCoc() {
+    const today = new Date();
+    const eightWeeksAgoDate = new Date();
+    var eightWeeksAgo = today.getDate() - 56;
+    eightWeeksAgoDate.setDate(eightWeeksAgo);
+    const defaultingStudents = await prisma.student.findMany({
+        where: {
+            createdAt: {
+                gt: new Date("2021-11-01")
```

Comment:
This changed to January, right @lsc0498 ?

Reply:
As an alternative, maybe we can comment this out (after testing it), so that we can toggle it on when it really should be launched ...

Reply:
It is basically a matter of setting up the CRON Job. So this is not really being executed at the moment.

Reply:
Ah, true, maybe we can then set the line to `gt: new Date("9999-01-01") // TODO: Set launch date when CoC enforcement is started` then so that setting this is not forgotten ...